oslibs/amigaos/include/stdint.h

```diff
@@ -1,4 +1,4 @@
-#ifdef __CLIB2__
+#if defined(__CLIB2__) || defined(__libnix__)
```

Comment:
Old libnix doesn't have stdint.h - how to handle that, I don't know.

Reply:
I first thought about using `__has_include`, but that is introduced in gcc5 - sigh.

Reply:
`__has_include` is supported as of gcc5, so not an option for gcc4.6

Reply:
I see, then don't have a good idea for a workaround for now.

Reply:
Does libnix v4 has a unique define that we can check against?

Reply:
I can't think of anything for libnix, but checking the __GNUC__ version could be a substitute.

Reply:
Looked at https://github.com/bebbo/libnix , but I can't find the full set of headers in there...

Reply:
I think most of the C headers come from ixemul.

Reply:
Most of the components used can be found in the main makefile:
https://github.com/bebbo/amiga-gcc/blob/master/Makefile#L36

Reply:
Ixemul.library is a bit of a mess, v50 and v60 were closed source forks, so many people stayed with v4x. Anyway, I'll try to figure out the root cause of the header clash.

Reply:
If it helps, this is the diff between stdint.h from bebbo's ixemul and ours:
```diff
--- ixemul-git/include/stdint.h
+++ uhexen2-git/oslibs/amigaos/include/stdint.h
@@ -1,12 +1,14 @@
+#ifdef __CLIB2__
+#include_next <stdint.h> /* clib2 provides a stdint.h already */
+
+#else
+
 #ifndef __STDINT_H
 #define __STDINT_H 1
 
+#include <machine/types.h> /* defines signed variants of int8_t..int64_t */
 #include <limits.h>
 
-typedef signed char int8_t;
-typedef signed short int16_t;
-typedef signed int int32_t;
-typedef signed long long int64_t;
 #define INT8_MIN SCHAR_MIN
 #define INT8_MAX SCHAR_MAX
 #define INT16_MIN SHRT_MIN
@@ -16,19 +18,19 @@
 #define INT64_MIN LLONG_MIN
 #define INT64_MAX LLONG_MAX
 
-typedef unsigned char uint8_t;
-typedef unsigned short uint16_t;
-typedef unsigned int uint32_t;
-typedef unsigned long long uint64_t;
+typedef u_int8_t uint8_t;
+typedef u_int16_t uint16_t;
+typedef u_int32_t uint32_t;
+typedef u_int64_t uint64_t;
 #define UINT8_MAX UCHAR_MAX
 #define UINT16_MAX USHRT_MAX
 #define UINT32_MAX UINT_MAX
 #define UINT64_MAX ULLONG_MAX
 
-typedef signed char int_least8_t;
-typedef signed short int_least16_t;
-typedef signed int int_least32_t;
-typedef signed long long int_least64_t;
+typedef int8_t int_least8_t;
+typedef int16_t int_least16_t;
+typedef int32_t int_least32_t;
+typedef int64_t int_least64_t;
 #define INT_LEAST8_MIN SCHAR_MIN
 #define INT_LEAST8_MAX SCHAR_MAX
 #define INT_LEAST16_MIN SHRT_MIN
@@ -38,19 +40,19 @@
 #define INT_LEAST64_MIN LLONG_MIN
 #define INT_LEAST64_MAX LLONG_MAX
 
-typedef unsigned char uint_least8_t;
-typedef unsigned short uint_least16_t;
-typedef unsigned int uint_least32_t;
-typedef unsigned long long uint_least64_t;
+typedef u_int8_t uint_least8_t;
+typedef u_int16_t uint_least16_t;
+typedef u_int32_t uint_least32_t;
+typedef u_int64_t uint_least64_t;
 #define UINT_LEAST8_MAX UCHAR_MAX
 #define UINTLEAST16_MAX USHRT_MAX
 #define UINTLEAST32_MAX UINT_MAX
 #define UINTLEAST64_MAX ULLONG_MAX
 
-typedef int8_t int_fast8_t;
-typedef int  int_fast16_t;
-typedef int int_fast32_t;
-typedef long long int_fast64_t;
+typedef int32_t int_fast8_t;
+typedef int32_t int_fast16_t;
+typedef int32_t int_fast32_t;
+typedef int64_t int_fast64_t;
 #define INT_FAST8_MIN INT_MIN
 #define INT_FAST8_MAX INT_MAX
 #define INT_FAST16_MIN INT_MIN
@@ -60,22 +62,22 @@
 #define INT_FAST64_MIN LLONG_MIN
 #define INT_FAST64_MAX LLONG_MAX
 
-typedef uint8_t uint_fast8_t;
-typedef unsigned int uint_fast16_t;
-typedef unsigned int uint_fast32_t;
-typedef unsigned long long uint_fast64_t;
+typedef u_int32_t uint_fast8_t;
+typedef u_int32_t uint_fast16_t;
+typedef u_int32_t uint_fast32_t;
+typedef u_int64_t uint_fast64_t;
 #define UINT_FAST8_MAX UINT_MAX
 #define UINT_FAST16_MAX UINT_MAX
 #define UINT_FAST32_MAX UINT_MAX
 #define UINT_FAST64_MAX ULLONG_MAX
 
 
-typedef int intptr_t;
-#define INTPTR_MIN INT_MIN
-#define INTPTR_MAX INT_MAX
+typedef long intptr_t;
+#define INTPTR_MIN LONG_MIN
+#define INTPTR_MAX LONG_MAX
 
-typedef unsigned int uintptr_t;
-#define UINTPTR_MAX UINT_MAX
+typedef unsigned long uintptr_t;
+#define UINTPTR_MAX ULONG_MAX
 
 typedef long long intmax_t;
 #define INTMAX_MIN LLONG_MIN
@@ -108,3 +110,5 @@
 #define UINTMAX_C(x) x##ULL
 
 #endif /* __STDINT_H */
+
+#endif /* __CLIB2__ */
```

Reply:
The problem appears to be sys/types.h which includes sys/_stdint.h. Fortunately that one has _..._T_DECLARED macro guards, so I can work this around properly in the oslibs stdint.h

Reply:
OK